### PR TITLE
Properly support per-system metadata file paths

### PR DESCRIPTION
**Pull request by WerWolv**
This PR fixes the long standing issue where ImHex was looking for files in subfolders next to the executable on all OSes. This is fine for Windows but causes a ton of issues on Mac and especially Linux.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,15 +113,13 @@ You need a C++20 compatible compiler such as GCC 10.2.0 to compile ImHex. Moreov
 - libmagic, libgnurx, libtre, libintl, libiconv
 - libcrypto
 - capstone
-- nlohmann json
 - Python3
 - freetype2
 - Brew (macOS only)
```
Comment (NicholeMattera) [inline README.md:118]: xCode is now also required for macOS.

File: README.md [hunk 2]
```diff
@@ -154,6 +150,47 @@ CC=$(brew --prefix llvm)/bin/clang CXX=$(brew --prefix llvm)/bin/clang++ PKG_CON
 make -j
 ```
 
+Install the ImHex executable as well as libimhex.dylib to wherever ImHex should be installed.
+
+All other files belong in `/Application Support/imhex`:
```
Comment (NicholeMattera) [inline README.md:155]: Should be `~/Application Support/imhex`

File: README.md [hunk 3]
```diff
@@ -154,6 +150,47 @@ CC=$(brew --prefix llvm)/bin/clang CXX=$(brew --prefix llvm)/bin/clang++ PKG_CON
...
+```
+Patterns: ~/Application Support/imhex/patterns
```
Comment (NicholeMattera) [inline README.md:157]: Same with all these. `~/Library/Application Support/...`

File: plugins/libimhex/source/helpers/utils_mac.mm [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    namespace hex {
+        std::string getPathForMac(ImHexPath path) {
+            @autoreleasepool {
+                NSError * error = nil;
+                NSURL * appSupportDir = [[NSFileManager defaultManager] URLForDirectory:NSApplicationSupportDirectory
+                                                                            inDomain:NSUserDomainMask
+                                                                    appropriateForURL:nil
+                                                                                create:YES
+                                                                                error:&error];
```
Comment (NicholeMattera) [inline plugins/libimhex/source/helpers/utils_mac.mm:14]: Not sure what happened with formatting here. (Pretty sure this was my fault.) Normally with Objective-C you line up the colons when doing multiple lines for a method invocations.

**Comment by WerWolv** [comment]
@NicholeMattera Thank you :) Should be good now

**Pull request merged**